workshop/exercise-7/README.md

```diff
@@ -40,67 +40,87 @@ NAME       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
 
-1. get the Helloworld pod name
+1. get the guesbook pod name
 
 ```sh
-kubectl get pods -l app=guestbook-ui
+kubectl get pods -l app=guestbook
 NAME                            READY     STATUS    RESTARTS   AGE
-guestbook-ui-596d68d88f-qxhzk   2/2       Running   1          1h
+guestbook-5fddf8db9c-5tk5v   2/2       Running   0          23m
+guestbook-5fddf8db9c-6m87p   2/2       Running   0          23m
+guestbook-5fddf8db9c-7nmcb   2/2       Running   0          23m
 ```
 
 Make sure the pod is “Running”.
 
 1. ssh into the envoy container
 
 ```sh
-kubectl exec -it guestbook-ui-xxxxxxxx -c istio-proxy /bin/bash
+kubectl exec -it guestbook-xxxxxxxx -c istio-proxy /bin/bash
 ```
 
 Make sure to change the pod name into the corresponding one on your system. This command will ssh into istio-proxy container(sidecar) of the pod.
 
 1. check out the certificate and keys are present
 
 ```sh
-ls /etc/certs/ 
+ls /etc/certs/
 ```
 
-You should see
+You should see the following (plus some others)
 
 ```sh
 cert-chain.pem   key.pem   root-cert.pem
 ```
 
 Note that `cert-chain.pem` is Envoy’s public certificate (i.e., presented to the peer), and `key.pem` is the corresponding private key. The `root-cert.pem` file is Istio Auth's root certificate, used to verify peers` certificates.
 
-1. send request to the guestbook-ui service
+### Disabling Authentication when Things Go Astray
+
+If, for some reason, mTLS is not functional, the Guestbook application will not operate correctly. This would be evident in its UI (e.g., it would display a [`Waiting for database connection`](waiting_on_database.png) message). In that case, mTLS should be disabled.
+
+mTLS it can be disabled globally or per service. 
+
+* To disable mTLS for all services in the mesh, comment out the `authPolicy` settings in the mesh configuration:
 
 ```sh
-curl https://guestbook-service:8080 -v --key /etc/certs/key.pem --cert /etc/certs/cert-chain.pem --cacert /etc/certs/root-cert.pem -k
+kubectl edit configmap istio -o yaml -n istio-system
+...
+apiVersion: v1
+data:
+  mesh: |-
+    # Uncomment the following line to enable mutual TLS between proxies
+    # authPolicy: MUTUAL_TLS
+    ...
 ```
 
-From the output there will be some error message `error fetching CN from cert:The requested data were not available`. This is expected.
-Go to the bottom and there you will see the success message.
+* To disable mTLS for connections originating from a specific service, add an `auth.istio.io/<port#>: NONE` annotation to the service definition:
 
 ```sh
-< HTTP/1.1 200 OK
-< x-application-context: helloworld-service
-< content-type: application/json;charset=UTF-8
-< date: Thu, 01 Feb 2018 06:12:46 GMT
-...
+kubectl edit service guestbook
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    auth.istio.io/3000: NONE
+  ...
```

Comment:
@elevran Did this option work?  I remember I had trouble and had to put the auth annotation NONE on analyzer to get it work.